auction-app-backend/src/main/java/com/example/auctionapp/service/implementation/ProductServiceImpl.java

```diff
@@ -237,6 +241,25 @@ public boolean hasActiveProducts(final UUID userId) {
     public void deleteActiveProducts(final UUID userId) {
         this.productRepository.deleteAllByUserEntity_UserIdAndStatus(userId, ProductStatus.ACTIVE);
     }
+    
+    @Override
+    public ProductPrices getProductPrices() {
+        final List<BigDecimal[]> pricesList = productRepository.findMinAndMaxPrices();
```

Comment:
Why BigDecimal and not double 
What do we get with BigDecimal that helps us that much to avoid using double?

Reply:
because when i was initially creating the product entity, i set the data type of price to be BigDecimal since as far as i know it provides more precision when using prices than double does, so i decided to stay consistent and use BigDecimal here as well